fitbenchmarking/results_processing/base_table.py

```diff
@@ -138,28 +135,51 @@ def get_colour(self, results):
         """
         Converts the result from
         :meth:`~fitbenchmarking.results_processing.base_table.Table.get_values()`
-        into the HTML colours
-        used in the tables. The base class implementation, for example,
-        uses the relative results and ``colour_scale`` within
+        into the HTML colours used in the tables. The base class
+        implementation, for example, uses the relative results and
+        ``colour_map``, ``colour_ulim`` and ``cmap_range`` within
         :class:`~fitbenchmarking.utils.options.Options`.
         :param results: tuple containing absolute and relative values
         :type results: tuple
         :return: dictionary containing HTML colours for the table
         :rtype: dict
         """
+        cmap_name = self.options.colour_map
+        cmap = plt.get_cmap(cmap_name)
+        cmap_ulim = self.options.colour_ulim
+        cmap_range = self.options.cmap_range
+        log_ulim = np.log10(cmap_ulim)  # colour map used with log spacing
         _, rel_value = results
         colour = {}
         for key, value in rel_value.items():
             if not all(isinstance(elem, list) for elem in value):
-                colour_index = np.searchsorted(self.colour_bounds, value)
-                colour[key] = [self.html_colours[i]
-                               for i in colour_index]
+                log_val = np.log10(value)
+                norm_val = (log_val - min(log_val)) /\
+                    (log_ulim - min(log_val))
+                norm_val[norm_val > 1] = 1  # applying upper cutoff
+                # trimming colour map according to default/user input
+                norm_val = cmap_range[0] + \
+                    norm_val*(cmap_range[1] - cmap_range[0])
+                rgba_list = cmap(norm_val)
+                hex_list = []
+                for c in rgba_list:
+                    hex_list.append(mpl.colors.rgb2hex(c))
+                colour[key] = hex_list
             else:
                 colour[key] = []
                 for v in value:
-                    colour_index = np.searchsorted(self.colour_bounds, v)
-                    colour[key].append([self.html_colours[i]
-                                        for i in colour_index])
+                    log_val = np.log10(v)
+                    norm_val = (log_val - min(log_val)) /\
+                        (log_ulim - min(log_val))
+                    norm_val[norm_val > 1] = 1  # applying upper cutoff
+                    # trimming colour map according to default/user input
+                    norm_val = cmap_range[0] + \
+                        norm_val * (cmap_range[1] - cmap_range[0])
+                    rgba_list = cmap(norm_val)
+                    hex_list = []
+                    for c in rgba_list:
+                        hex_list.append(mpl.colors.rgb2hex(c))
+                    colour[key].append(hex_list)
```

Comment:
There's duplicated code here which made me pause for a second. I think it would probably be better to pull this out to a seperate function. e.g.:
```
@staticmethod
def _val_to_colour(vals):
    """
    Converts values to hexadecimal colour strings.

    :param vals: values in the range [0, 1] to convert to colour strings
    :type vals: list[float]

    :return: colours as hex strings for each input value
    :rtype: list[str]
    """
    ...
```

Reply:
I have introduced a method `_val_to_colour(vals, cmap, cmap_range, log_ulim)` in my latest push. This has reduced the repetition significantly so thanks for the suggestion. I couldn't get the `@staticmethod` to work, it gave the following error `TypeError: 'staticmethod' object is not callable` so it is just a method at the moment.